deployer/src/deployer/search/__init__.py

```diff
@@ -40,42 +40,32 @@ def index(
     if not update:
```

Comment:
I don't really know what `Document._index` returns, so I'm probably off here, but I assume it returns the index for the day (i.e., `mdn_docs_2021mmdd`)? If we're in `update` mode, it seems that `document_index` will be the new one (based on today's date) and not the previous one that we want to update? Also, if we're **not** in `update` mode, we can't run this more than once a day, or else we'll delete the same index the alias is pointing to, right?

Reply:
No, I don't think that's a concern. 
Suppose that we create a brand new index at 00:00 on mdn_docs_20210401 and we populate it. Then at 01:00 we use `--update` and "reindex" (aka. cake on) another 100 documents, they will update `mdn_docs_20210401` which is fine because Kuma will always search on the alias which, for the rest of the day is pointing to `mdn_docs_20210401`. 

The alias just means you're searching in `mdn_docs` which is always the last index to be populated independent of the day. 

But imagine if we "copied" mdn_docs_20210401 into mdn_docs, then mdn_docs would be missing the new additions if you go back to populate more into mdn_docs_20210401. But aliases aren't copies. They're just aliases. 

Lastly...

> if we're not in update mode, we can't run this more than once a day, or else we'll delete the same index the alias is pointing to?

Yes, it's the truth. If you run `poetry run deployer search-index /path/to/build` TWICE in one day, it will create a downtime window. It's almost as if we're saying "I change my mind. I want to start over for today's index. Delete the old one and fill it all back up again."
But this is not going to happen outside of local development. Once we start adding partial builds throughout the day, the timeline will be:

- Monday morning: `poetry run deployer search-index /path/to/build` (will delete and populate a new index)
- Monday lunch: `poetry run deployer search-index /path/to/build --update` (will append to Monday's index)
- Monday afternoon: `poetry run deployer search-index /path/to/build --update` (will append to Monday's index)
- Tuesday morning: `poetry run deployer search-index /path/to/build` (will delete and populate a new index)
- Tuesday lunch: `poetry run deployer search-index /path/to/build --update` (will append to Tuesday's index)
- Tuesday afternoon: `poetry run deployer search-index /path/to/build --update` (will append to Tuesday's index)

We currently have `prod-build.yml` which we execute every 24h. 
But hopefully, we'll (soon) have `prod-partial-build.yml` which builds based on `git` history several times a day but then it'll use `poetry run deployer search-index /path/to/build --update` on those.

Reply:
Assuming `Document._index` returns the index for the day (i.e., `mdn_docs_2021mmdd`) I still think this logic is confusing (i.e., there are hidden assumptions that make it fragile). It seems to me that if `--update` is specified, we want to ask the alias for its current index, and use that one, instead of calling `Document._index`. In other words, as the code is currently written, it seems that `--update` only works if you do it on the same day that you already created an index. So, if you create a new index on Monday, don't do anything else until Tuesday, and on Tuesday only try to do an update, it will fail. I guess I'm saying that `--update` feels fragile because we're not asking the alias for its current index and using that, which seems to me what we should do when `--update` is specified.

Also, for the case when we're not using `--update`, we're assuming that we're only going to do that once per day. If we do that more than once, we'll have downtime. Why not use the time of day (to the minute or even second) as part of the index name? That way we can run a full reindex multiple times a day (if we want) with no downtime.

I guess I'm saying that the current implementation assumes a daily usage pattern (only one full reindex per day, and updates only work for that day) that isn't clear from the CLI options, and with some minor adjustments, we can eliminate those assumptions.